extensions/amp-selector/1.0/amp-selector.js

```diff
@@ -283,9 +298,12 @@ function OptionShim({
     if (!onClick) {
       return;
     }
-    shimDomElement.addEventListener('click', onClick);
+    shimDomElement.addEventListener('click', (e) => onClick(e));
```

Comment:
The issue is that `AmpSelector.Component = SelectorShim;` and other props should instead be `AmpSelector['Component'] = ...`